File: Frontend/lib/models/comment.dart
class Comment {
  final String avatarUrl;
  final String userName;
  final String date;
  final int stars;
  final String comment;

  Comment({
    required this.avatarUrl,
    required this.userName,
    required this.date,
    required this.stars,
    required this.comment,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      avatarUrl: "https://i.pravatar.cc/150?img=2",
      userName: json['createdBy'],
      date: json['createdAt'],
      stars: json['star'],
      comment: json['content'],
    );
  }
}

File: Backend/src/models/Comment.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const dataSchema = new mongoose.Schema({
    content: {
        required: true,
        type: String
    },
    createdBy: {
        required: true,
        type: Schema.Types.ObjectId, 
        ref: 'user' 
    },
    createdAt: {
        required: true,
        type: Date,
        default: Date.now
    },
    postId: {
        required: true,
        type: Schema.Types.ObjectId, 
        ref: 'post' 
    },
    star: {
        required: true,
        type: Number,
        default: 0,
        min: 0,
        max: 5
    }
})

module.exports = mongoose.model('comment', dataSchema)